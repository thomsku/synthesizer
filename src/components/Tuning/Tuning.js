import React from "react";
import Radio from "@material-ui/core/Radio";
import RadioGroup from "@material-ui/core/RadioGroup";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import FormLabel from "@material-ui/core/FormLabel";
import PropTypes from "prop-types";
import "./Tuning.css";

const Tuning = ({
  setTuning,
  tuning,
}) => {
  return (
    <div className="radiobuttons">
      <FormLabel className="parametertitle" component="legend">Tuning</FormLabel>
      <RadioGroup
        aria-label="Tuning"
        name="tuning"
        className="radiogroup"
        value={tuning}
        onChange={setTuning}
      >
        <FormControlLabel className="radiolabel" value="just" control={<Radio className="radio"/>} label="just intonation" />
        <FormControlLabel className="radiolabel" value="equal" control={<Radio className="radio"/>} label="equal temperament" />

      </RadioGroup>
    </div>
  );
};

Tuning.propTypes = {
  setTuning: PropTypes.func.isRequired,
  tuning: PropTypes.string.isRequired,
};

export default Tuning;
