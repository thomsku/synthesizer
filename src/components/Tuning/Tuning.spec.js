import React from "react";
import {shallow} from "enzyme";
import Tuning from "./Tuning";

describe("Tuning", () => {
  let wrapper;
  beforeEach(() => wrapper = shallow(<Tuning setTuning={jest.fn()} tuning="just"/>));

  it("should render correctly", () => expect(wrapper).toMatchSnapshot());

  it("should render a <div />", () => {
    expect(wrapper.find("div")).toHaveLength(1);
  });
});
