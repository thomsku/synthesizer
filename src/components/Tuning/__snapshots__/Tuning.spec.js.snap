// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tuning should render correctly 1`] = `
<div
  className="tuningRadioButtons"
>
  <WithStyles(WithFormControlContext(ForwardRef(FormLabel)))
    component="legend"
  >
    Tuning
  </WithStyles(WithFormControlContext(ForwardRef(FormLabel)))>
  <ForwardRef(RadioGroup)
    aria-label="Tuning"
    className="RadioGroup"
    name="tuning"
    onChange={[MockFunction]}
    value="just"
  >
    <WithStyles(WithFormControlContext(ForwardRef(FormControlLabel)))
      className="radiolabel"
      control={
        <ForwardRef(WithStyles)
          className="radio"
        />
      }
      label="Just Intonation"
      value="just"
    />
    <WithStyles(WithFormControlContext(ForwardRef(FormControlLabel)))
      className="radiolabel"
      control={
        <ForwardRef(WithStyles)
          className="radio"
        />
      }
      label="Equal Temperament"
      value="equal"
    />
  </ForwardRef(RadioGroup)>
</div>
`;
